### ISSUE TITLE
perf(solve): NewtonSolver.solve re-compiles on every call (jit closure built per call)

### ISSUE THREAD

**Issue by ctdegroot** [DeGrootResearchGroup/aquaflux#102]
## Summary

`NewtonSolver.solve` applies `@eqx.filter_jit` to a closure **defined inside the method**, so every
call constructs a new jitted function with a fresh compilation cache. Any caller that invokes
`solve()` in a loop — the segregated turbulence driver being the motivating case — pays a full
re-trace **and XLA re-compile on every call**, forever. The compiled code is never reused.

`aquaflux/solve/newton.py:132`:

```python
def solve(self, residual_fn, phi0):
    @eqx.filter_jit          # <- new function object on every solve() call
    def step(phi):
        return newton_step(residual_fn, phi, solver=self.solver, preconditioner=self.preconditioner)

    phi = phi0
    for _ in range(self.iterations):
        phi = step(phi)
    return phi
```

## Evidence

Measured on the momentum residual of the periodic law-of-the-wall channel (480 cells, 3 Newton
iterations per solve), calling `solve()` four times with an updated effective viscosity — i.e. the
segregated driver's access pattern. Compiles counted via `jax.log_compiles()`:

| call | current (closure per call) | hoisted jit, Module as argument |
|---|---|---|
| 0 | 1.29 s (1 compile) | 2.08 s (1 compile) |
| 1 | 0.93 s (1 compile) | **0.19 s (0 compiles)** |
| 2 | 0.92 s (1 compile) | **0.33 s (0 compiles)** |
| 3 | 1.09 s (1 compile) | **0.22 s (0 compiles)** |

The current path recompiles on every call and never warms up; the fixed path compiles once. Roughly
**4x per call** at 480 cells, and the gap widens with mesh size because compile cost scales with the
graph, not the data.

The same signature shows up end-to-end. Profiling the law-of-the-wall case (8 sweeps, 480 cells),
the per-sweep flow solve is flat at `1.89 1.66 1.71 1.63 1.70 1.65 1.73 1.84` s — no warm-up — while
the scalar `k` solve in the same run drops `0.7 s -> 0.06 s` once its cache hits. The flow solve was
54% of total wall time.

## Why hoisting the decorator is not sufficient

`residual_fn` arrives as a bare callable, and callers pass a bound method of a freshly-built
assembler each sweep (the driver rebuilds `momentum` via `_with_viscosity` -> `eqx.tree_at`).
Equinox treats a non-Module callable as static and hashes it by identity, so a hoisted
`@eqx.filter_jit` would still miss the cache every sweep.

For the cache to hit, the residual has to travel as **data** — an `equinox.Module` passed as an
argument, so its array leaves are dynamic and only its structure is static.

## Precedent already in the tree

Both correct patterns exist and can be followed rather than invented:

- `ImplicitNewtonSolver` (`aquaflux/solve/implicit.py`) contains **no internal jit** at all; it
  leaves jitting to the caller. `NewtonSolver` is the outlier here.
- `reused_flow_solve` (`aquaflux/flow/continuation.py:186`) wraps a **single** hoisted
  `@eqx.filter_jit` around a solve taking `MomentumContinuity` as an argument, and its docstring
  already describes this exact failure mode ("because each is a new object, retraces and recompiles
  the whole solve — a per-sweep cost that grows with mesh size").

## Options

1. **Drop the internal jit** and let callers own the jit boundary, matching `ImplicitNewtonSolver`.
   Simplest and most consistent; makes un-jitted use slower (each Newton step dispatched eagerly),
   so callers that currently rely on the implicit jit need updating in the same change.
2. **Take the residual as a Module.** Change the signature to `solve(residual_fn, phi0, params)` with
   `residual_fn(phi, params)`, mirroring `ImplicitNewtonSolver.solve(lambda s, m: m.residual(s), state, momentum)`,
   and hoist one jit to construction. Fixes caching within the class at the cost of a public API change.
3. **Jit the whole iteration loop** rather than each step. Independent of 1/2 and worth doing either
   way: currently `iterations=15` means 15 separate XLA executions with a host round-trip between
   each, where one fused executable would do.

Option 1 or 2 needs a decision; option 3 composes with both. Pre-release, so the API break is free.

## Affected call sites

`NewtonSolver(...).solve(...)` inside a loop or repeated call:

- `tests/integration/test_channel_law_of_the_wall.py:80` (per outer sweep — the motivating case)
- `aquaflux/flow/initialization.py:156`
- `tests/integration/test_betchen_solve.py` (lines 75, 97, 111, 128)
- `tests/integration/test_poiseuille.py` (lines 57, 77), `tests/integration/test_plane_wall.py` (lines 90, 147)

Single-shot callers are unaffected in correctness and only pay the one compile they would anyway.

## Not in scope

Purely a compile-caching/performance defect — results are numerically unchanged. Separate from the
scalar k/omega convergence work in #99.

**Comment by ctdegroot** [comment]
Stale — closing. **The class this is about no longer exists.**

`NewtonSolver` was deleted; `aquaflux/solve/newton.py` now exposes only `newton_step` / `newton_correction` (one correction each), and the nonlinear path is `ImplicitNewtonSolver`. `grep -rn "class NewtonSolver" aquaflux` returns nothing, so the per-call `@eqx.filter_jit` closure described here has no host left.

The deletion was made for a stronger reason than this performance defect: the class was `newton_step` plus a **fixed-count, unchecked loop**, which is redundant at `iterations=1` (what nearly every call site used, a linear residual being exact in one correction) and wrong above it, since a fixed count cannot distinguish convergence from exhaustion. The rule that replaced it: a *linear* residual takes `newton_step`, a *nonlinear* one takes `ImplicitNewtonSolver`.

Neither survivor has the defect. `newton_step` does not jit internally at all — the caller owns the jit boundary — and `ImplicitNewtonSolver`'s march is a `lax.while_loop` inside one `custom_vjp`, compiled once.
